utils/utils.go

```diff
@@ -462,15 +457,29 @@ func customizeNetworkPolicyPorts(ingress *networkingv1.NetworkPolicyIngressRule,
 		currentLen = desiredLen
 	}
 
-	// Add additional ports needed
+	// Add additional ports if needed
 	for currentLen < desiredLen {
 		ingress.Ports = append(ingress.Ports, networkingv1.NetworkPolicyPort{})
 		currentLen++
 	}
 
-	for i, port := range ports {
-		newPort := &intstr.IntOrString{Type: intstr.Int, IntVal: port}
-		ingress.Ports[i].Port = newPort
+	// Initialize the normal port
+	primaryPortIntVal := ba.GetService().GetPort()
+	if ba.GetService().GetTargetPort() != nil {
+		primaryPortIntVal = *ba.GetService().GetTargetPort()
+	}
+	primaryPort := intstr.IntOrString{Type: intstr.Int, IntVal: primaryPortIntVal}
+	ingress.Ports[0].Port = &primaryPort
+
+	// Initialize additional ports
+	for i := 0; i < len(ba.GetService().GetPorts()); i++ {
+		portPtr := &ba.GetService().GetPorts()[i]
+		if portPtr.TargetPort.String() != "" {
```

Comment:
`String()` function could return `"<nil>"`, not sure whether that's something we should check for

Reply:
I think the TargetPort property is not a pointer https://pkg.go.dev/k8s.io/api/core/v1#ServicePort so it would have a default value initialized for elements of the struct.

Reply:
I think that's correct, ServicePort.TargetPort  is not a point so it can not be nil (defaults to 0, Int type) it might be initialized to equal port.  It might be that `if portPtr.TargetPort.String() != ""` might be always `true` though.

Reply:
Okay. Sounds good. I'll go ahead and merge. Thank you